fix: Mandatory VRF request fees estimation

akhercha opened the pull request:
<!-- Reference any GitHub issues resolved by this PR -->

Closes #NA

## Introduced changes

<!-- A brief description of the changes -->
- Re-added mandatory fee check
- Multicall impl (without wait_for_tx)
- whitelist for which we don't check fees

Note: Adding `track_nonce` adds way to much unreliability. Better to loose a few seconds waiting for the tx than pilling up txs that will fail and retry requests.

##

- [ ] This PR contains breaking changes

<!-- List of all breaking changes -->

EvolveArt commented on pragma-sdk/pragma_sdk/onchain/mixins/randomness.py:
```diff
@@ -509,14 +477,16 @@ async def handle_random(
-                f" {hex(invoke_tx.transaction_hash)}"
+                f" {hex(invoke_tx.hash)}"
             )
+        print(f"Send tx took: {(time.time()) - NOW:02f}s")
```
same

akhercha replied:
done

EvolveArt commented on vrf-listener/benchmark/stress/stress_tester.py:
```diff
@@ -68,7 +68,7 @@ async def run_stresstest(self) -> None:
         users = await self.config.get_users_client(
             network=self.network,
-            rpc_url=self.rpc_url,
+            rpc_url="https://api.cartridge.gg/x/starknet/sepolia",
```
revert this change ? rpc should be configurable no?

akhercha replied:
done

akhercha commented on pragma-sdk/pragma_sdk/onchain/mixins/randomness.py:
```diff
@@ -166,50 +164,20 @@ async def submit_random_multicall(
             ]
         )
         all_calls = flatten_list(all_calls)
+        print(f"Call creation took: {(time.time()) - ESS:02f}s")
```
done

akhercha commented:
@EvolveArt 

Also added a `whitelist` param to the vrf-listener.
We can now add whitelisted addresses for which we won't check fees & pay if necessary for the callback.

Also added in the `infra` - just need to add an env variable `LOOT_CONTRACT_ADDRESS`
